doc/design/upgrade.md

```diff
@@ -0,0 +1,118 @@
+# Upgrade
+
+After installed VCH, VIC admin would want to upgrade existing instances to latest version. This document describes the upgrade requirement, and design options.
+
+## Requirements
+
+### Scope
+
+While a new version is ready, new version's vic-machine upgrade can upgrade one running or powered off VCH appliance. The containers managed by this VCH will be left in existing version. To have a new version container, user need to create new container from upgraded VCH instance. If existing container is stopped and restarted, the old version should be kept.
+
+VIC version includes three parts, "release tag"-"build id"-"github commit hash".
+
+vic-machine upgrade will rely on build id change only. User need to provide newer binary with bigger build id, to upgrade existing VCH.
+
+We could also use github commit hash for version diff, but that will be difficult to check if one version is newer than another one, so at this time, we'll leverage build id only. This requires our build system to keep increase build id no matter any system change. 
+
+vic-machine upgrade cannot upgrade VCH to a version newer than itself.
+
+### Impact
+
+The VCH will have downtime during upgrade, suppose container lifecyle is not closely coupled with VCH up time, for VIC has good network isolation. User could run appliacation and connect to their container without VCH is running. But there still have some impact, including docker attach, docker log and docker port mapping, which can be resumed after upgrade is finished.
```

Comment:
The VCH _appliance_ (control plane) will have downtime during the upgrade. Container lifecycles are not coupled with that of the appliance and, when using mapped vsphere networks (vic-machine _--container-network_ argument) instead of port forwarding, the network data paths are not dependent on the appliance either.
There will be impact on container interaction while the appliance is down:
* no direct access to container logs
* no attach ability
* network outage for NAT based port forwarding
All of those facets should resume normal operation after upgrade is complete